extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts

```diff
@@ -1,8 +1,34 @@
-export default (id,value,viewport) => {
+/**
+ * Sets the viewport zoom/pan information on initial display.
+ * 
+ * @param id 
+ * @param value 
+ * @param viewport 
+ */
+const setViewportZoomPan = (id,value,viewport) => {
   const {zoom,pan, flipHorizontal, flipVertical} = value;
   if( zoom && zoom!==1 ) viewport.setZoom(zoom);
   if( pan ) viewport.setPan(pan);
   if( flipHorizontal || flipVertical ) {
     viewport.flip(value);
   }
-}
\ No newline at end of file
+}
+
+/** Stores the viewport zoom/pan information, when a viewport goes away (to be restored later) */
+const storeViewportZoomPan = (settings, priorSettings, viewport) => {
```

Comment:
Store the previous viewports values here - so that this code is NOT contained in core OHIF, which makes it far cleaner.